manifests/resource/location.pp

```diff
@@ -153,66 +153,65 @@
-  $location_custom_cfg_prepend = undef,
-  $location_custom_cfg_append  = undef,
-  $include                     = undef,
-  $try_files                   = undef,
-  $proxy_cache                 = false,
-  $proxy_cache_key             = undef,
-  $proxy_cache_use_stale       = undef,
-  $proxy_cache_valid           = false,
-  $proxy_method                = undef,
-  $proxy_http_version          = undef,
-  $proxy_set_body              = undef,
-  $proxy_buffering             = undef,
-  $auth_basic                  = undef,
-  $auth_basic_user_file        = undef,
-  $auth_request                = undef,
-  $rewrite_rules               = [],
-  $priority                    = 500,
-  $mp4                         = false,
-  $flv                         = false,
-  $expires                     = undef,
+  Optional[String] $proxy                             = undef,
+  Optional[String] $proxy_redirect                    = $::nginx::proxy_redirect,
+  String $proxy_read_timeout                          = $::nginx::proxy_read_timeout,
+  String $proxy_connect_timeout                       = $::nginx::proxy_connect_timeout,
+  Array $proxy_set_header                             = $::nginx::proxy_set_header,
+  Array $proxy_hide_header                            = $::nginx::proxy_hide_header,
+  Array $proxy_pass_header                            = $::nginx::proxy_pass_header,
+  Optional[String] $fastcgi                           = undef,
+  Optional[String] $fastcgi_index                     = undef,
+  Optional[Hash] $fastcgi_param                       = undef,
+  String $fastcgi_params                              = "${::nginx::conf_dir}/fastcgi_params",
+  Optional[String] $fastcgi_script                    = undef,
+  Optional[String] $fastcgi_split_path                = undef,
+  Optional[String] $uwsgi                             = undef,
+  Optional[Hash] $uwsgi_param                         = undef,
+  String $uwsgi_params                                = "${nginx::config::conf_dir}/uwsgi_params",
+  Optional[String] $uwsgi_read_timeout                = undef,
+  Boolean $ssl                                        = false,
+  Boolean $ssl_only                                   = false,
+  Optional[String] $location_alias                    = undef,
+  Optional[Enum['any', 'all']] $location_satisfy      = undef,
+  Optional[Array] $location_allow                     = undef,
+  Optional[Array] $location_deny                      = undef,
+  Optional[Boolean ] $stub_status                     = undef,
+  Optional[Variant[String, Array]] $raw_prepend       = undef,
+  Optional[Variant[String, Array]] $raw_append        = undef,
+  Optional[Hash] $location_custom_cfg                 = undef,
+  Optional[Hash] $location_cfg_prepend                = undef,
+  Optional[Hash] $location_cfg_append                 = undef,
+  Optional[Hash] $location_custom_cfg_prepend         = undef,
+  Optional[Hash] $location_custom_cfg_append          = undef,
```

Comment:
Possibly, but maybe not. See https://github.com/voxpupuli/puppet-nginx/issues/1100#issuecomment-319899019